19.md

```diff
@@ -4,21 +4,38 @@ NIP-19
 bech32-encoded entities
 -----------------------
 
-`draft` `optional` `author:jb55` `author:fiatjaf`
+`draft` `optional` `author:jb55` `author:fiatjaf` `author:Semisol`
+
+This NIP species all bech32-encoded entities.
+
+## Bare keys
 
 To prevent confusion and mixing between private keys, public keys and event ids, which are all 32 byte strings. bech32-(not-m) encoding with different prefixes can be used for each of these entities.
 
-For public keys, the prefix is `npub`, so, for example, `3bf0c63fcb93463407af97a5e5ee64fa883d107ef9e558472c4eb9aaaefa459d` translates to `npub180cvv07tjdrrgpa0j7j7tmnyl2yr6yr7l8j4s3evf6u64th6gkwsyjh6w6`.
+These are the possible bech32 prefixes:
+
+  - `npub`: public keys
+  - `nsec`: private keys
+  - `note`: note ids
+
+Example: the hex public key `3bf0c63fcb93463407af97a5e5ee64fa883d107ef9e558472c4eb9aaaefa459d` translates to `npub180cvv07tjdrrgpa0j7j7tmnyl2yr6yr7l8j4s3evf6u64th6gkwsyjh6w6`.
+
+## Shareable identifiers with extra metadata
 
-For private keys, the prefix is `nsec`.
+When sharing a profile or an event, an app may decide to include relay information and other metadata such that other apps can locate and display these entities more easily.
 
-For kind-1 notes, the prefix is `note`.
+For these events, the contents are a binary-encoded list of `TLV` (type-length-value), with `T` and `L` being 1 byte each (`uint8`, i.e. a number in the range of 0-255), and `V` being a sequence of bytes of the size indicated by `L`.
 
-## Optional extra metadata in bech32-encoded blobs
+These are the possible bech32 prefixes:
```

Comment:
These are the possible bech32 prefixes with `TLV`:

I think it makes sense to speficy that these are the ones with a TLV, so it's a different line than from above..